src/frontend/sharedComponents/Images/CoreBlobImage.tsx

```diff
@@ -0,0 +1,67 @@
+import {useAttachmentUrl} from '@comapeo/core-react';
+import {type ComponentProps, useState} from 'react';
+
+import {PhotoVariant} from '../../sharedTypes';
+import {GracefulImage} from './GracefulImage';
+import {ImageErrorPlaceholder} from './ImageErrorPlaceholder';
+
+type Props = {
+  driveId: string;
+  name: string;
+  projectId: string;
+} & Omit<ComponentProps<typeof GracefulImage>, 'renderError' | 'onLoadError'>;
+
+export const CoreBlobImage = ({
+  driveId,
+  name,
+  projectId,
+  testID,
+  ...otherImageProps
+}: Props) => {
+  const [displayedVariant, setDisplayedVariant] =
+    useState<PhotoVariant>('original');
+
+  const {
+    data: attachmentUrl,
+    error,
```

Comment:
does using this error actually work? i'm under the impression that in the case of the initial query failing, the use of a suspense hook here is actually going to propagate to the nearest error boundary (for which we have none!!!)

Reply:
just tested this by setting the "variants" to fetch to some nonsense and it works as intended! kind of surprised because i thought react query would've attempted to trigger an error boundary 🤔

Reply:
came to the conclusion that attempting to use the `error` here doesn't work as intended in the case of the initial query resulting in an error. Ended up moving the error handling to the consuming component and using error boundaries. see https://github.com/digidem/comapeo-mobile/pull/1179/commits/a660e64bd62f50a8318adf439093cd694d2e61ef

for future reference, a quick way to force test suspense errors is to implement a bad query in the component. for example, I added this to the component that uses the core-react hooks:

```ts
const badQuery = useSuspenseQuery({ queryKey: ['bad'], queryFn: () => throw new Error('Failed') })
```

This allowed me to test how the error boundaries I added are used.